modules/Components/src/client/ACASThingBrowser.coffee

```diff
@@ -111,27 +86,92 @@ class ThingSimpleSearchController extends AbstractFormController
 					@$(".bv_thingSearchTerm").attr "disabled", false
 					@$(".bv_doSearch").attr "disabled", false
 
+	getQueryTerms: (queryTerms, searchTerm) ->
+		for queryValue in @configs
+			#Default is all display values are searchable
+			isSearchable = !queryValue.isSearchable? || queryValue.isSearchable != false
+			# Code Name is part of the default search so skip it here
+			if isSearchable && queryValue.key != "codeName"
+				if queryValue.key == "recordedDate"
+					@addRecordedDateToQuery(queryTerms, searchTerm)
+				else
+					#If the key is an ls value
+					valDef = @model.getValueInfo(queryValue.key)
+					if valDef?
+							searchOperator = "~"
+							if queryValue.searchOperator?
+								searchOperator = queryValue.searchOperator
+							queryTerms.queryDTO.values.push	
+								stateType: valDef.stateType
+								stateKind: valDef.stateKind
+								valueType: valDef.type
+								valueKind: valDef.kind
+								operator: searchOperator
+					else
+						# If the key is an ls label
+						labDef = @model.getLabelInfo(queryValue.key)
+						if labDef?
+							searchOperator = "~"
+							if queryValue.searchOperator?
+								operator = queryValue.searchOperator
+							queryTerms.queryDTO.labels.push	
+								labelType: labDef.type
+								labelKind: labDef.kind
+								operator: searchOperator
+		return queryTerms
+
+	addRecordedDateToQuery: (queryTerms, searchTerm) ->
+		# Default search for recordedDate is ISO without time
+		# e.g. 2021-06-02
+		dateParts = searchTerm .split('-')
+		# Offset the user entered date by one month to account for month = 0 in javascript
+		if typeof(dateParts[1]) != "undefined"
+			dateParts[1] = dateParts[1]-1
```

Comment:
Little worried here if I type in dash-separated terms that are not dates. Will this bomb out if I type in `brian-bolt-rules` ?

Reply:
Yeah, understandable and so was I.  Essentially most cases a user enters will not return a valid date and therefore won't be added to the query because there is a check on the date (`!isNaN(recordedDateGreaterThan))`)

Some examples:
`1234-1234-1234`: Pared into, `Mon Feb 15 1340 16:07:02 GMT-0752` and added to the query (no results because its a crazy date)
`brian-bolt-rules`: Invalid date, not added to query
`2021-1-Monday`: Invalid date, not added to query

In other parts of code we do test for a "reasonable date" but I did not add that here becaues unreasonable dates are unlikely to have real matching values so I didn't think it would hurt much.

Reply:
I think this stems from my lack of understanding of Javascript. As long as line 131:
```recordedDateGreaterThan = new Date(Date.UTC(...dateParts))```
just produces `recordedDateGreaterThan = undefined` or NaN, then the code makes sense to me as-is. Was just confused that we cast to date without any try/catch.